### PR TITLE
Tokenize FAB Types

### PR DESCRIPTION
**Pull request by ayush547**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [x] android

### Description of changes
ButtonCore was designed to be type shared across Button and FAB, but states (like 'pressed') were given the type of ButtonTokens, this would have allowed FAB 'pressed' to have 'primary' and other Button states (and not in FAB) inside it.

Button and FAB on Android have an additional requirement of an inner border, have added the corresponding types.
Added required states for FAB.
Removed 'hovered' from FAB since it is designed to be mobile component.

Minor Fix: FABCore root slot was set to be View instead of Pressable.

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by rurikoaraki** [comment]
Could you make sure the SPEC files are updated to include the new tokens & type changes?

File: packages/components/Button/SPEC.md [hunk 1]
```diff
@@ -191,9 +202,9 @@ export interface ButtonTokens extends LayoutTokens, FontTokens, IBorderTokens, I
   iconWeight?: number;
 
   /**
-   * The width of the button.
+   * An object describing the shadow of the button
```
Comment (ankraj12) [inline packages/components/Button/SPEC.md:205]: hyper nit: end with "." - all comments

File: packages/components/Button/src/CompoundButton/SPEC.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 The `CompoundButton` is a `Button` that is designed to show an extra line of text.
 
+Does not render on Android.
```
Comment (ankraj12) [inline packages/components/Button/src/CompoundButton/SPEC.md:7]: nits: remove the extra line above. If not already, run prettier across the change

File: packages/components/Button/src/FAB/SPEC.md [hunk 3]
```diff
@@ -36,39 +41,51 @@ Android:
...
-  content?: string;
-
   /*
    * Source URL or name of the icon to show on the Button.
    */
   icon?: IconSourcesType;
 
+  /**
+   * Button contains only icon, there's no text content
+   * Must be set for button to style correctly when button has not content.
```
Comment (ankraj12) [inline packages/components/Button/src/FAB/SPEC.md:70]: *no content .

File: packages/components/Button/src/Button.types.ts [hunk 4]
```diff
@@ -53,6 +53,18 @@ export interface ButtonCoreTokens extends LayoutTokens, FontTokens, IBorderToken
    */
   shadowToken?: ShadowToken;
 
+  /**
+   * Focused State on Android has inner and outer borders
+   * Outer Border is equivalent to the border tokens from IBorders
+   * This adds inner border support.
```
Comment (ankraj12) [inline packages/components/Button/src/Button.types.ts:59]: This line is redundant, can be removed - the names are explanatory.

**Review by ankraj12 — APPROVED** [review]

**Pull request merged**